add opt-in  non_empty_function rule (#5617)
realm/SwiftLint: CHANGELOG.md

* Linting got around 20% faster due to the praisworthy performance
  improvements done in the [SwiftSyntax](https://github.com/apple/swift-syntax)
  library.

> * Add `no_empty_block` default rule to validate that code blocks should not be empty.

Reviewer: Please add an empty line above.

> * Add `no_empty_block` default rule to validate that code blocks should not be empty.

Reviewer: ```suggestion
* Add `no_empty_block` default rule to validate that code blocks are not empty.
```

>   They should at least contain a comment.

Reviewer: ```suggestion
  They should at least contain a comment.  
```

Please append two spaces to the end of the description. They are required to enforce a line break in the rendered Markdown.

* Linting got around 20% faster due to the praisworthy performance
  improvements done in the [SwiftSyntax](https://github.com/apple/swift-syntax)
  library.

> *  Add `no_empty_block` default rule to validate that `function`, `init`, `deinit`, `catch` and `defer` should not have empty code block. They should at least contain a comment.

Reviewer 2: Probably want to wrap this text the same as the rest of the file
